woocommerce/woocommerce-gutenberg-products-block
# Fix products incorrectly marked as discounted

## What

We are incorrectly displaying some products as discounted when their price is updated to add some extra cost.

Fixes #10433

## Why

Revised the `isDiscounted` logic. Now, the discount is applied only when the new price is lower than the `regularPrice`, rather than just being different.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Add a discounted item and a regular-priced item to your cart.
2. Go to the Cart Block page. Confirm that the discount label is only visible next to the discounted product.
3. Go to the Checkout Block page. Under the `Order Summary` section, confirm that the discount label is only visible next to the discounted product.
4. Add the following code snippet using the [Code Snippets](https://wordpress.org/plugins/code-snippets/) plugin:

```PHP
function calc_price( $cart_object ) {
    foreach ( $cart_object->get_cart() as $hash => $value ) {
        $value['data']->set_price( 50 );
    }
}
add_action( 'woocommerce_before_calculate_totals', 'calc_price' );
```

5. Go to the Cart Block page. Confirm that both products reflect the new price (50) and neither displays a discount label.
6. Go to the Checkout Block page, under the `Order summary` section, Confirm that both products reflect the new price (50) and neither displays a discount label.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|    <img width="798" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/14235870/368da5d5-4bdc-4f7c-a3c9-4128bcf9f0c9">    |   <img width="751" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/14235870/0c8295a9-1088-4cfd-ba5b-ab9c874e8d71">   |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Fix products incorrectly marked as discounted
